### ISSUE TITLE
W2.1: Freeze benchmark config YAML

### ISSUE THREAD

**Issue by freddysongg** [freddysongg/LLM-W#6]
## PRD

**Problem:** Benchmark across three backends requires bit-identical training config on each run to be a valid controlled comparison. Any drift invalidates the results.

**Goal:** Single committed YAML config file used verbatim by all three backend runs, with SHA256 hash recorded in each run's DB row.

## Technical Spec

Create `configs/bench/qwen15b-lora.yaml`. Config contents:

```yaml
model:
  name: Qwen/Qwen2.5-1.5B-Instruct
  revision: main  # pin via git SHA before first bench run
dataset:
  name: HuggingFaceH4/ultrachat_200k
  subset_size: 2000
  seed: 42
training:
  epochs: 1
  batch_size: 4
  seq_len: 512
  learning_rate: 2.0e-4
  lr_scheduler: cosine
  warmup_ratio: 0.03
adapters:
  type: lora
  rank: 16
  alpha: 32
  dropout: 0.05
  target_modules: [q_proj, k_proj, v_proj, o_proj]
quantization:
  # Applied only on CUDA backend; ignored on MPS
  enabled_on_cuda: true
  load_in_4bit: true
  bnb_4bit_quant_type: nf4
  bnb_4bit_compute_dtype: bfloat16
eval:
  split_hash: # populated by #9
  perplexity_samples: 200
```

Backend-aware behavior: quantization applied only when `device=cuda`. MPS path ignores the quantization block. This is the only device-branched behavior.

## Acceptance Criteria

- [ ] `configs/bench/qwen15b-lora.yaml` committed
- [ ] Model revision pinned to a specific git SHA (not `main`) before first run
- [ ] `runs.config_hash` column populated with SHA256 of this file for every bench run
- [ ] Three bench runs (M1 Pro, M4 Pro, A10) produce identical `config_hash` in SQLite
- [ ] Config validates against existing `WorkbenchConfig` Pydantic schema

## Dependencies

- **Blocked by:** none
- **Blocks:** #7 (runner needs config to reference), #9 (split hash gets injected here)

## References

- Spec: `docs/plans/2026-04-13-llm-w-improvement-plan-v4.md` §W2.1
- Existing schema: `backend/app/schemas/workbench_config.py`

**Comment by freddysongg** [comment]
Shipped in 81457be.

**Shape translation**: Spec YAML in the ticket body used simplified key names (`model.name`, `dataset.name`, flat training) that don't match the real `WorkbenchConfig` Pydantic schema. Translated to the canonical 13-section shape so the AC check "validates against WorkbenchConfig" holds. Verified:

```
python3 -c "...WorkbenchConfig.model_validate(yaml.safe_load(...))"
→ project=bench-qwen15b-lora, model=Qwen/Qwen2.5-1.5B-Instruct, max_samples=2000, lora r/α=16/32, quant mode=4bit
```

**Values locked per spec**: epochs=1, batch=4, seq_len=512, lr=2e-4, cosine, warmup 0.03, LoRA r=16 α=32 dropout=0.05 targets=[q,k,v,o]_proj, 4bit nf4 bfloat16, seed=42, subset 2000.

**Device branching for quantization**: Schema has no per-device conditional. Runner (#7) will flip `quantization.enabled` to `false` when `--device mps`/`cpu` before constructing the trainer. Documented in an inline comment in the YAML.

**Bench sidecar** (`bench:` top-level section): ignored by WorkbenchConfig (Pydantic extra='ignore' default), consumed only by `scripts/bench/run_local.sh`. Holds `eval_split_hash` (populated by #9), `perplexity_samples: 200`, and `judge_sanity_prompts_path` (used by #10).

**Config hash (SHA256)**: `ed67ad6b89efd7e8c14780926e25c5e9e788f3d124b141164d3eb6566f2df1ce` — this is what every bench run's `runs.config_hash` column must equal. Will change when the `TODO(ws2)` pin of `model.revision` from `main` to a specific SHA lands before the first real bench run.

AC status:
- [x] configs/bench/qwen15b-lora.yaml committed
- [ ] Model revision pinned to SHA — deferred to first-run prep
- [ ] runs.config_hash populated for every bench run — depends on #8 column landing and #7 runner
- [ ] Three bench runs produce identical config_hash — will be verified when runs exist
- [x] Config validates against WorkbenchConfig Pydantic schema
